src/test/ui/nll/named-region-basic.stderr

```diff
@@ -1,31 +0,0 @@
-warning: not reporting region error due to -Znll
-  --> $DIR/named-region-basic.rs:19:5
-   |
-19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
-   |     ^^^
-
-error[E0597]: `*x` does not live long enough
-  --> $DIR/named-region-basic.rs:19:6
-   |
-19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
-   |      ^^ does not live long enough
-   |
-   = note: borrowed value must be valid for the static lifetime...
-note: ...but borrowed value is only valid for the lifetime 'a as defined on the function body at 18:1
-  --> $DIR/named-region-basic.rs:18:1
-   |
-18 | / fn foo<'a, 'b>(x: &'a u32, y: &'b u32) -> &'b u32 {
-19 | |     &*x //~ ERROR free region `'a` does not outlive `'b`
-20 | |     //~^ ERROR `*x` does not live long enough
-21 | |     //~| WARN not reporting region error due to -Znll
-22 | | }
-   | |_^
-
-error: free region `'a` does not outlive `'b`
-  --> $DIR/named-region-basic.rs:19:5
-   |
-19 |     &*x //~ ERROR free region `'a` does not outlive `'b`
-   |     ^^^
-
-error: aborting due to 2 previous errors
-
```

Comment:
Thinking about the tests, I didn't see a test where a free region was directly compared against a bound region. aka
```Rust
fn foo<'a, F: for<'x> FnOnce(
    Cell<&'a u32>,
    Cell<&'x u32>)>(_cell: Cell<&'a u32>, _f: F) { }

fn case1() {
    let a = 0;
    let cell = Cell::new(&a);
    foo(cell, |cell_1, cell_2| {
        cell_1.set(cell_2.get()); // forces 'x: 'a, error in closure
    })
}

fn case1() {
    let a = 0;
    let cell = Cell::new(&a);
    foo(cell, |cell_1, cell_2| {
        cell_2.set(cell_1.get()); // forces 'a: 'x, implies 'a = 'static -> borrow error
    })
}
```

Because there

Reply:
You could also have a test where we can use a given to convert a free/bound relationship to a free/free relationship, but we already have tests that show we can use givens to convert bound/bound to free/free (propagate-despite-same-free-region) so I think it's less critical to add *another* test - you can still add one if you want :-).

Reply:
yeah, I think existing tests cover that ok (free/bound -> free/free).